app/scripts/services/deployments.js

```diff
@@ -268,6 +268,13 @@ angular.module("openshiftConsole")
       return annotation(object, 'deployment.kubernetes.io/revision');
     };
 
+    DeploymentsService.prototype.getName = function(object) {
```

Comment:
I'd rather have one function that returns both the log resource and the resource name (perhaps as an object). The way this is written, this assumes too much about what `getLogResource` will return.

Reply:
They seem decoupled to me, but  I'm pretty sure its ignorance on my part.  Let me take a stab & see if I follow:

- `getName()` alone seems standalone, but perhaps not very useful.
- `getLogResource()` seems to require the output of `getName()` to actually be useful.

So I could update the return value of `getLogResource()` to `{ resource: rgv, name: getName(obj)  }`.

Reply:
They're coupled in that you can't change one without changing the other. If we ever return something other than `DeploymentConfig` in getLogResource for replication controllers owned by a deployment config, then you have to change `getName` too or it will return the wrong value. We more of less have that same logic in two places.

Better to keep them as one method to avoid that problem.

> I could update the return value of getLogResource() to { resource: rgv, name: getName(obj) }.

Yeah, that's what I was thinking.